datamodel.customer-survey.xml

```diff
@@ -1880,8 +1880,29 @@ EOF
 		{
 			$sBody = $this->Get('email_body');
 		}
-		$sBody .= '<br/><a href="'.$sQuizzUrl.'">'.Dict::S('Survey-notif-linktoquizz').'</a>';
-
+		
+	
+		$aParams = [];
+		$aParams['recipient->object()'] = $oContact;
+		$aParams['quiz_url'] = $sQuizzUrl;
+		
+		$oContactOnBehalfOf = MetaModel::GetObject('Contact', $this->Get('on_behalf_of'), false, true);
+		if(!is_null($oContactOnBehalfOf)) {
+			$aParams['on_behalf_of->object()'] = $oContactOnBehalfOf;
+		}
+		
+		// Email body contains a place holder
+		$bPrependLink = true; // classic behavior before introduction of placeholders
+		if(preg_match('/\$quiz_url\$/', $sBody)) {
+			$bPrependLink = false;
+		}
```

Comment:
```suggestion
		// Email body contains a placeholder?
		$bPrependLink = !preg_match('/\$quiz_url\$/', $sBody);
```

Reply:
The suggestion is included in the current code. Closing conversation.

Reply:
Indeed, I couldn't mark as resolved. I think only author and maintainers can?

Reply:
I think so !